packages/p2p/src/components/my-ads/create-ad-add-payment-method-modal.jsx

```diff
@@ -81,16 +76,24 @@ const CreateAdAddPaymentMethodModal = () => {
                     {localize('Add payment method')}
                 </React.Fragment>
             }
-            toggleModal={onCancel}
+            toggleModal={e => {
+                if (
+                    !e.target ||
+                    ['svg', 'use'].includes(e.target.tagName) || // sometimes the svg has not loaded yet
+                    ['dc-dropdown-list__item', 'dc-btn', 'dc-icon'].every(
+                        className => !e.target.className.includes(className)
+                    )
```

Comment:
```suggestion
                    ['dc-dropdown-list__item', 'dc-btn', 'dc-icon'].every(
                        className => !e.target.className.includes(className)
                    )
```
why are we checking the logic based on class names?

Reply:
This was an issue with the `Modal` component, where if there is another 'floating' element like the `Autocomplete` dropdown list, it will detect clicking the dropdown items as clicking outside the modal and call `toggleModal` (even though within the `Modal` component it already queried `.dc-dropdown-list` to check as portal elements), and passing `elements_to_ignore` does not work since the dropdown list does not appear once the user clicks on the dropdown form itself

Reply:
Sometimes it also detects icons besides the X button to be clicking outside the modal as well 😢

Reply:
I tried debugging the following scenarios:
[x] Using `elements_to_ignore` to query `dc-dropdown-list__item`, it returns undefined
[x] Memoizing the modal component with `React.memo`, since I noticed that it re-renders multiple times before evaluating if the user is clicking outside with `useOnClickOutside`
[x] Setting portal id for dropdown list